spec.html

```diff
@@ -758,12 +758,18 @@ <h1>InitializeTypedArrayFromTypedArray ( _O_, _srcArray_ )</h1>
           1. Let _bufferConstructor_ be ? SpeciesConstructor(_srcData_, %ArrayBuffer%).
         1. Else,
           1. Let _bufferConstructor_ be %ArrayBuffer%.
+        1. <ins>Let _data_ be ? AllocateArrayBuffer(_bufferConstructor_, _byteLength_).</ins>
+        1. <ins>If IsIntegerIndexedObjectOutOfBounds(_srcArray_, _getSrcBufferByteLength_) is *true*, throw a *TypeError* exception.</ins>
+        1. <ins>Set _elementLength_ to IntegerIndexedObjectLength(_srcArray_, _getSrcBufferByteLength_).</ins>
+        1. <ins>Set _byteLength_ to _elementSize_ &times; _elementLength_.</ins>
```

Comment:
The new _byteLength_ might be greater than the old _byteLength_. _byteLength_ should only be updated if the buffer was shrunk. Alternatively, you could assign the new byte length to a _newByteLength_, and pass min(_newByteLength_, _byteLength) to CopyDataBlockBytes.

Reply:
Persisting the variable `_byteLength_` might be confusing in what is becoming a somewhat complicated algorithm (readers might not recognize that despite its authoritative name, its value is not the byte length). There's also `_elementLength_` to consider, which maybe ought to be complimented with a corresponding `_newElementLength_` (more for coherence than a direct need).

I think we can use the new buffer's [[ArrayBufferByteLength]] slot to get the prior value of `_byteLength_` without having to communicate "newness" or "oldness" through variable names. I've pushed up a commit to demonstrate what I have in mind. Does that seem appropriate to you?

Reply:
We can't use [[ArrayBufferByteLength]] got growable `SharedArrayBuffer`s, which stores its byte length in a separate shared block. That's why the idempotent getter exists.